apps/dashboard/src/app/login/page.tsx

```diff
@@ -38,14 +39,39 @@ export default function LoginPage() {
 function CustomConnectEmmbed() {
   const isLG = useMediaQuery("(min-width: 1024px)");
   const searchParams = useSearchParams();
+  const router = useDashboardRouter();
   const { theme } = useTheme();
+  const nextSearchParam = searchParams?.get("next");
+
+  function onLoginSuccessful() {
+    if (nextSearchParam && isValidRedirectPath(nextSearchParam)) {
+      router.replace(nextSearchParam);
+    } else {
+      router.replace("/dashboard");
+    }
+  }
+
   return (
     <ConnectEmbed
       auth={{
         getLoginPayload,
-        doLogin: (params) => doLogin(params, searchParams?.get("next")),
+        doLogin: async (params) => {
+          try {
+            await doLogin(params);
+            onLoginSuccessful();
+          } catch (e) {
+            console.error("Failed to login", e);
+            throw e;
+          }
+        },
         doLogout,
-        isLoggedIn,
+        isLoggedIn: async (x) => {
+          const isLoggedInResult = await isLoggedIn(x);
+          if (isLoggedInResult) {
+            onLoginSuccessful();
+          }
+          return isLoggedInResult;
+        },
       }}
       client={thirdwebClient}
```

Comment:
We used to have a similar logic on the middleware - but that was simply checking for the existence of auth cookie - which is not enough to determine if the user is actually logged in 

Other alternative is to do the proper isLoggedIn check on the middleware if the user lands on /login page